pkg/apis/pipeline/v1beta1/param_context.go

```diff
@@ -130,16 +134,26 @@ func getContextParams(ctx context.Context, overlays ...Param) []Param {
 		p := Param{
 			Name: ps.Name,
 		}
-		if ps.Type == ParamTypeString {
+		switch ps.Type {
+		case ParamTypeString:
 			p.Value = ArrayOrString{
 				Type:      ParamTypeString,
 				StringVal: fmt.Sprintf("$(params.%s)", ps.Name),
 			}
-		} else {
+		case ParamTypeArray:
 			p.Value = ArrayOrString{
 				Type:     ParamTypeArray,
 				ArrayVal: []string{fmt.Sprintf("$(params.%s[*])", ps.Name)},
 			}
+		default:
+			objVal := make(map[string]string)
+			for k := range ps.Properties {
+				objVal[fmt.Sprintf("$(params.%s.%s)", p.Name, k)] = ""
```

Comment:
> Is it possible to pass through the whole object instead of each field individually? (thinking ahead to if/when objects are a mix of strings/arrays/objects)

sgtm!

As `ObjectVal` is a map, how about this - having only one entry in the `ObjectVal` and passing through the whole object by using the param name as the entry's key and using the whole object as the entry's value. Does it make sense?